recipes/gemma_full_finetune_distributed.py

```diff
@@ -0,0 +1,520 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import sys
+import time
+from dataclasses import dataclass, field
+
+from functools import partial
+from typing import Any, Dict, Optional, Tuple
+from warnings import warn
+
+import torch
+from omegaconf import DictConfig
+
+from torch import nn
+from torch.distributed import init_process_group
+from torch.distributed.fsdp import (
+    FullOptimStateDictConfig,
+    FullStateDictConfig,
+    FullyShardedDataParallel as FSDP,
+    StateDictType,
+)
+from torch.distributed.fsdp.wrap import ModuleWrapPolicy
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader, DistributedSampler
+
+from torchtune import config, modules, utils
+
+from torchtune.recipe_interfaces import FTRecipeInterface
+
+from tqdm import tqdm
+
+
+log = utils.get_logger("DEBUG")
+
+
+@dataclass
```

Comment:
You don't need to define this hear. This can just be added to the yaml file? Just add a section with:

```
share_weights: True
share_weights_mapping: {
   "output.weight" : "tok.weight"
}
```

I think this should work. And then you can read this directly from the yaml